### PR TITLE
Package the godot engine

### PR DESCRIPTION
**Pull request by alexlarsson**
This is https://godotengine.org/, an open source game engine editor.

### CODE REVIEW

**Comment by alexlarsson** [comment]
Test build at https://flathub.org/builds/#/builders/1/builds/385

File: org.godotengine.Godot.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                {
+                    "type": "archive",
+                    "url": "http://downloads.sourceforge.net/project/scons/scons/2.5.1/scons-2.5.1.tar.gz",
+                    "sha256": "0b25218ae7b46a967db42f2a53721645b3d42874a65f9552ad16ce26d30f51f2"
+                }
+            ],
+            "build-commands": [ "python setup.py install --prefix=/app" ]
+        },
+        {
+            "name": "glu",
```
Comment (TingPing) [inline org.godotengine.Godot.json]: glu is in shared-modules.

File: org.godotengine.Godot.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>org.godotengine.Godot.desktop</id>
+  <name>Godot</name>
+  <summary>Godot game engine editor</summary>
+  <description>
+    Godot is an advanced, feature-packed, multi-platform 2D and 3D open source game engine.
+  </description>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT</project_license>
```
Comment (TingPing) [inline org.godotengine.Godot.appdata.xml]: Should add `<releases>` so software has a useful version number.

File: org.godotengine.Godot.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    "tags": [],
+    "finish-args": [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* Network access */
+        "--share=network",
+        /* Sound access */
+        "--socket=pulseaudio",
+        /* Wayland access */
+        "--socket=wayland",
```
Comment (TingPing) [inline org.godotengine.Godot.json]: Seems like wayland isn't supported: https://github.com/godotengine/godot/issues/576

**Comment by alexlarsson** [comment]
Also seems like there is no aarch64 support

File: org.godotengine.Godot.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        /* We want full fs access so we can read the files */
+        "--filesystem=host",
+        "--device=dri"
+    ],
+    "build-options" : {
```
Comment (Barthalion) [inline org.godotengine.Godot.json]: Shouldn't we fallback to upstream CFLAGS? If not, wouldn't it make more sense to have these enforced by flatpak-builder (even with some command line parameter) so we could stop duplicating it everywhere?

Comment (alexlarsson) [reply org.godotengine.Godot.json]: This is a tricky area. I like the build result to be repeatable, but having the cflags as a commandline means you will not get the same build from two runs. Also, such a commandline would just be a default, we would still require it to be possible to override for some modules, and we have no great way of doing that for a globally set thing atm.

It is kind of painful to have to repeat this all over the place though.

I'm not sure there is a great "default" in upstream, i think it just does nothing. Will check.

**Comment by alexlarsson** [comment]
New test build: https://flathub.org/builds/#/builders/1/builds/394

**Comment by alexlarsson** [comment]
Merged this.

**Pull request closed without merging**